demos/TransferLearning/TransferLearningGui.py

```diff
@@ -0,0 +1,219 @@
+        ax[i // 3, i % 3].axis('off')
+        ax[i // 3, i % 3].set_title("Predicted:{}".format(predicted_class), color='r')
+
+        if predicted_class.lower() in demo.test_generator.filenames[wrong_guess_indices[i]]:
+            ax[i // 3, i % 3].set_title("Predicted:{}".format(predicted_class), color='g')
+    print('Initial Prediction:')
+    display(plt)
+
+    demo.train(model, backend, epochs, batch_size, [progress_info])
+
+    # Update prediction with trained model
+    updated_prob = demo.predict([progress_info])
+    updated_pred = np.argmax(updated_prob, axis=1)
+
+    # Display updated guesses
+    f, ax = plt.subplots(3, 3, figsize=(15, 15))
+
+    for i in range(0, 9):
+        test_image = os.path.join(demo.test_dir,
+                                  demo.test_generator.filenames[wrong_guess_indices[i]])
+        imgRGB = mpimg.imread(test_image)
+        predicted_class = "Dog" if updated_pred[wrong_guess_indices[i]] else "Cat"
+
+        ax[i // 3, i % 3].imshow(imgRGB)
+        ax[i // 3, i % 3].axis('off')
+        ax[i // 3, i % 3].set_title("Predicted:{}".format(predicted_class), color='r')
+
+        if predicted_class.lower() in demo.test_generator.filenames[wrong_guess_indices[i]]:
+            ax[i // 3, i % 3].set_title("Predicted:{}".format(predicted_class), color='g')
+    print('Updated Prediction:')
+    display(plt)
+
+    train_button.disabled = False
+
+
+clear_output()
+
+out = widgets.Output(layout={'border': '1px solid black'})
+
+model_dropdown = widgets.Dropdown(options=['ResNet50', 'MobileNet v2'],
+                                  value='ResNet50',
+                                  description='Model:')
+ngraph_dropdown = widgets.Dropdown(options=ngraph_backends,
+                                   value=ngraph_backends[0],
+                                   description='nGraph:')
+batch_slider = widgets.IntSlider(min=1, max=100, value=batch_size, description='Batch Size:')
+epoch_slider = widgets.IntSlider(min=1, max=16, value=epochs, description='Epochs:')
+
+train_button = widgets.Button(description='Train', disabled=False)
+
+training_tab = widgets.VBox(
+    children=[model_dropdown, ngraph_dropdown, batch_slider, epoch_slider, train_button])
+#fine_tuning_tab = widgets.VBox(children=[
+#        batch_slider, epoch_slider])#, fine_tune_slider, fine_tune_button, classify_button,
+#        #shuffle_button])
+#fine_tuning_tab.disabled = True
+tab = widgets.Tab(children=[training_tab])  #, fine_tuning_tab])
+tab.set_title(0, 'Training')
+#tab.set_title(1, 'Fine Tuning')
+
+progress_bar = widgets.IntProgress()
+progress_text = widgets.Label()
+epoch_text = widgets.Label()
+progress_box = widgets.HBox([progress_bar, progress_text, epoch_text])
+
+gui = widgets.VBox(children=[tab, progress_box])
+
+display(gui)
+
+model_dropdown.observe(on_model_change)
+ngraph_dropdown.observe(on_ngraph_change)
+train_button.on_click(on_train_clicked)
+#self.shuffle_button.on_click(self.init_model)
```

Comment:
All of this code gets run when? It's hard to tell because it will happen whenever it gets imported. It's better to put this all in a function that will be explicitly called when you want it to be called.